Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -194,6 +194,39 @@ extension Channel {
         }
     }
 
+    /// Configures a `ChannelPipeline` to speak HTTP/2 and sets up mapping functions so that it may be interacted with from concurrent code.
+    ///
+    /// In general this is not entirely useful by itself, as HTTP/2 is a negotiated protocol. This helper does not handle negotiation.
+    /// Instead, this simply adds the handler required to speak HTTP/2 after negotiation has completed, or when agreed by prior knowledge.
+    /// Use this function to setup a HTTP/2 server pipeline if you wish to use async sequence abstractions over inbound and outbound streams, as it allows that pipeline to evolve without breaking your code.
+    ///
+    /// - parameters:
+    ///     - mode: The mode this pipeline will operate in, server or client.
+    ///     - connectionConfiguration: The settings that will be used when establishing the connection. These will be sent to the peer as part of the
+    ///         handshake.
+    ///     - streamConfiguration: The settings that will be used when establishing new streams. These mainly pertain to flow control.
+    ///     - streamDelegate: The delegate to be notified in the event of stream creation and close.
+    ///     - position: The position in the pipeline into which to insert this handler.
+    ///     - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
+    /// - returns: An `EventLoopFuture` containing the `AsyncThrowingStream` of inbound HTTP/2 stream channels and  the `AsyncStreamMultiplexer` inserted into this pipeline, which can be used to initiate new streams.
+    @_spi(AsyncChannel)
+    public func configureHTTP2PipelineAsync<InboundStreamChannelType>(mode: NIOHTTP2Handler.ParserMode,
+                                       connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration,
+                                       streamConfiguration: NIOHTTP2Handler.StreamConfiguration,
+                                       streamDelegate: NIOHTTP2StreamDelegate? = nil,
+                                       position: ChannelPipeline.Position = .last,
+                                       inboundStreamInitializer: @escaping @Sendable (Channel) -> EventLoopFuture<InboundStreamChannelType>) -> EventLoopFuture<(AsyncThrowingStream<InboundStreamChannelType, any Error>, NIOHTTP2Handler.AsyncStreamMultiplexer)> {
```

Comment:
I was kind of expecting a different return type here.
1. We should create our own concrete type like `NIOHTTP2InboundStreamChannels` which holds the `AsyncThrowingStream` under the hood. This allows us to change the underlying root async sequence in the future.
2. We should create a concrete type that holds both the `NIOHTTP2InboundStreamChannels` and the `AsyncStreamMultiplexer`
3. I am wondering if we want to expose the `AsyncStreamMultiplexer` at all or rather just a method on the new type that allows you to create a new outbound stream channel
4. We probably have to return the connection channel itself as well since we are adding handlers to it and changing the inbound and outbound types there. We might want to wrap that into a `NIOAsyncChannel` unconditionally.

Reply:
> We can't use `any` here since we have to support older versions. Just drop the `any` for now

We can use `any`: it was added in 5.6 which is our minimum supported version. (https://github.com/apple/swift-evolution/blob/main/proposals/0335-existential-any.md)

Reply:
> I was kind of expecting a different return type here.
> 
> 1. We should create our own concrete type like `NIOHTTP2InboundStreamChannels` which holds the `AsyncThrowingStream` under the hood. This allows us to change the underlying root async sequence in the future.
> 2. We should create a concrete type that holds both the `NIOHTTP2InboundStreamChannels` and the `AsyncStreamMultiplexer`
> 3. I am wondering if we want to expose the `AsyncStreamMultiplexer` at all or rather just a method on the new type that allows you to create a new outbound stream channel

Agree.

> 4. We probably have to return the connection channel itself as well since we are adding handlers to it and changing the inbound and outbound types there. We might want to wrap that into a `NIOAsyncChannel` unconditionally.

We are already holding the connection channel (this is an extension on `Channel`) so we don't need to return it. I also don't think we should return `NIOAsyncChannel` _here_, that will come later. This is a lower-level building block which most users will never use.

Reply:
Makes sense!

Reply:
I have made these changes but had the `AsyncStreamMultiplexer` hold the `NIOHTTP2InboundStreamChannels` so that we continue with the tradition of being the object with which we create outbound channels.